src/optimizer/child_property_deriver.cpp

```diff
@@ -95,18 +96,15 @@ void ChildPropertyDeriver::Visit(UNUSED_ATTRIBUTE const Aggregate *op) {
 void ChildPropertyDeriver::Visit(const Limit *op) {
   // Limit fulfill the internal sort property
   std::vector<PropertySet *> child_input_properties{new PropertySet()};
-  auto provided_prop = new PropertySet();
 
   // Limit must satisfy output sort properties but child can attempt to satisfy sort property optionally
   if (!op->GetSortExpressions().empty()) {
     const std::vector<common::ManagedPointer<parser::AbstractExpression>> &exprs = op->GetSortExpressions();
     const std::vector<catalog::OrderByOrderingType> &sorts{op->GetSortAscending()};
-    provided_prop->AddProperty(new PropertySort(exprs, sorts));
-
-    child_input_properties[0]->AddProperty(new PropertySort(exprs, sorts, false));
+    child_input_properties[0]->AddProperty(new PropertySort(exprs, sorts));
   }
 
-  output_.emplace_back(provided_prop, std::move(child_input_properties));
+  output_.emplace_back(new PropertySet(), std::move(child_input_properties));
```

Comment:
I think that if we push the sort down to the child, then the `Limit` should still have an output sort property.

Reply:
Hmm, how come? Wouldn't the Limit's sort already be fulfilled by the child since it gets pushed down?

Reply:
So maybe I'm have some misunderstanding with how this function behaves, but since the sort is fulfilled by the child and the Limit node doesn't break the sort, then it will also have a sorted output.  Otherwise `OptimizeExpressionCostWithEnforcedProperty` will see that the `Limit` node has a required sort property, but no output sort property and will enforce it with an OrderBy node.

Reply:
The required properties from the `OptimizationContext` will only be enforced if they are not in the output properties derived from this function (Lines 340 of optimizer_task.cpp). So if a Limit has a required sort property, which it will if there's an order by, but it doesn't have an output sort property then it will generate an OrderBy node. Since the child will have already satisfied the sort property, this OrderBy node is unnecessary.

Reply:
Yeah, you're 100% right, my bad

Reply:
Fixed